kore/test/Test/Kore/Step/Axiom/Matcher.hs

```diff
@@ -862,6 +876,119 @@ test_unificationWithAppMatchOnTop =
             (Mock.f Mock.a)
             (Mock.f Mock.b)
         assertEqualWithExplanation "" expect actual
+    , testCase "handles ambiguity" $ do
```

Comment:
Is the ambiguity here that the right-hand side of the `\ceil`-simplification below branches?

Reply:
Yes, that's the only way to generate ambiguity for now. I added a comment, not sure it helps.